src/celeritas/neutron/xs/NeutronElasticMacroXsCalculator.hh

```diff
@@ -0,0 +1,86 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/neutron/xs/NeutronElasticMacroXsCalculator.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Assert.hh"
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/cont/Span.hh"
+#include "celeritas/UnitTypes.hh"
+#include "celeritas/mat/MaterialView.hh"
+
+#include "NeutronElasticMicroXsCalculator.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Calculates the macroscopic cross section.
+ */
+class NeutronElasticMacroXsCalculator
```

Comment:
Good idea as I will also add a similar set of calculators for `NeutronInelastic`.  I think that a templated class, something like,
```
template<class MicroXsT, typename MacroXsUnitsT>
class MacroXsCalculator
```
may be used as a common method.  As a quick look, this requires changes in the associated `MicroXsCalculator` (argument for constructor and operator) and a creation of a specific `MicroXsCalculator` before hand which imposes an explicit rule for `MacroXsCalculator`.  Please let me know if you have the better idea or if a potential consolidation may be a separated MR.

Reply:
I like that idea. I think it would be better to create a new PR that refactors LivermorePEMacroXsCalculator to something like:
```c++
template<class F>
class MacroXsCalculator
{
    using ParamsRef = typename F::ParamsRef;
    using Energy = typename F::Energy;
    using XsUnits = units::Native;

    MacroXsCalculator(ParamsRef const&, MaterialView const& material);
    real_type operator()(Energy) const;
};
```

I think also `LivermorePEMicroXsCalculator` should be returning either a `units::Barn` quantity or a native (len^2) unit.